exp/controllers/azuremachinepoolmachine_controller.go

```diff
@@ -110,6 +111,15 @@ func (ampmr *AzureMachinePoolMachineController) SetupWithManager(ctx context.Con
 		return errors.Wrapf(err, "failed adding a watch for AzureMachinePool model changes")
 	}
 
+	// Add a watch on CAPI Machines for MachinePool Machines
+	if err := c.Watch(
+		source.Kind(mgr.GetCache(), &clusterv1.Machine{}),
+		handler.EnqueueRequestsFromMapFunc(util.MachineToInfrastructureMapFunc(infrav1exp.GroupVersion.WithKind("AzureMachinePoolMachine"))),
```

Comment:
is this going to filter only Machines owned by an AzureMachinePool?

Reply:
Yes (implicitly) because it filters only for Machines that have an infraRef pointing to this specific AzureMachinePoolMachine.